### ISSUE TITLE
autoloader_layout test overly dependant on the name of the current directory

### ISSUE THREAD

**Issue by ghost** [rodjek/puppet-lint#245]
About a year ago, b91d0058da6265ade231bfcd5516bb0050ecc3d5 added a leading '/' to the expected path in the _autoloader_layout_ test.  This only works if the module that you're testing is in a directory with the same name as the module itself.  It seems like this should not fail, but it does:

```
$ git clone https://github.com/puppetlabs/puppetlabs-ntp.git
$ cd puppetlabs-ntp
$ puppet-lint manifests/init.pp
manifests/init.pp:1:autoloader_layout:ERROR:ntp not in autoload module layout'
```

The commit that I referenced earlier fixed the problem that it described, but created this new issue.

The following test demonstrates the problem:

``` ruby
context 'foo::bar in rodjek-foo/manifests/bar.pp' do
  let(:code) { "class foo::bar { }" }
  let(:path) { 'rodjek-foo/manifests/bar.pp' }

  it 'should not detect any problems' do
    expect(problems).to have(0).problems
  end
end
```
